src/Microsoft.AspNetCore.Mvc.TagHelpers/CacheTagHelper.cs

```diff
@@ -147,27 +149,37 @@ public CacheTagHelper(IMemoryCache memoryCache, HtmlEncoder htmlEncoder) : base(
         // Internal for unit testing
         internal MemoryCacheEntryOptions GetMemoryCacheEntryOptions()
         {
+            var hasEvictionCriteria = false;
             var options = new MemoryCacheEntryOptions();
             if (ExpiresOn != null)
             {
+                hasEvictionCriteria = true;
                 options.SetAbsoluteExpiration(ExpiresOn.Value);
             }
 
             if (ExpiresAfter != null)
             {
+                hasEvictionCriteria = true;
                 options.SetAbsoluteExpiration(ExpiresAfter.Value);
             }
 
             if (ExpiresSliding != null)
             {
+                hasEvictionCriteria = true;
                 options.SetSlidingExpiration(ExpiresSliding.Value);
             }
 
             if (Priority != null)
             {
+                hasEvictionCriteria = true;
```

Comment:
I don't think `Priority` should be considered an eviction criteria. Looking at https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.memorycacheentryoptions

It doesn't specify when something should be evicted, only its relative importance.

Also we removed the memory-pressure-based automatic eviction, so this won't happen without someone triggering it.